src/components/analyticVisualization/VisualizationGroup/VisualizationGroup.js

```diff
@@ -26,10 +27,38 @@ const findDefaultGroup = (type, groupList, settings) => {
     return defaultGroup
 }
 
-export const VisualizationGroup = ({ settings, onChange, groupList, type }) => {
-    const { data: id } = useSystemId()
+const isDefaultGroup = group => group.name === 'default'
+
+const findGroup = (type, group, settings) => {
+    const { groupList, groupId } = group
+
+    switch (type) {
+        case 'program':
+        case 'dataset':
+            return settings.group
+        default:
+            return find(groupList, g => g.id === groupId)
+    }
+}
+
+export const VisualizationGroup = ({ settings, onChange, groupList, type, groupId, disabled }) => {
+    const { refetch: refetchId, data: id } = useSystemId()
     const [group, setGroup] = useState(true)
     const [groupType, setGroupType] = useState(true)
+    const [title, setTitle] = useState('')
+
+    useEffect(() => {
+        refetchId()
+        const groupFound = findGroup(type, { groupList, groupId }, settings)
+        if (disabled) {
+            if (!isDefaultGroup(groupFound)) {
+                setGroup(!!groupFound)
+                setTitle(groupFound.name)
+            } else {
+                setGroup(false)
+            }
+        }
+    }, [groupList])
```

Comment:
Hey,
I am curious to know if `type`, `groupId`, `disabled`, and `settings` can be part of the dependency array. When their values change the `useEffect` hook should run and compute again?
Thanks!

Reply:
Hey, I'm using the `useEffect` hook to:
- refetch the id: the random id I get from that will be assigned if a new group has been created.
- In case I'm editing the visualization title: to validate if the visualization belongs to a group and assign the name/title of this group 

If I add `type`, `groupId`, `disabled`, and `settings` to the dependency array, then the id will be generated multiple times. I need only one new id each time. What do you think?

Reply:
Maybe a `useEffect` with an empty dependency array because I only need one id each time.